src/Installer/Installer/Base.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Dock\Installer\Installer;
+
+use Dock\Installer\InstallContext;
+use Dock\Installer\InteractiveProcessRunner;
+use Dock\IO\UserInteraction;
+use SRIO\ChainOfResponsibility\ChainBuilder;
+
+abstract class Base
```

Comment:
Instead of this installer abstract class, you can inject a `TaskProvider` that will be different based on the OS.